.github/workflows/codecov.yml

```diff
@@ -0,0 +1,30 @@
+name: CodeCov
+on:
+    workflow_run:
+        workflows: [Tests]
+        types: [completed]
+
+jobs:
+    coverage_report:
+        runs-on: ubuntu-latest
+        if: ${{ github.event.workflow_run.conclusion == 'success' }}
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v3
+
+            - name: Install dependencies
+              run: |
+                  python -m pip install --upgrade pip
+                  pip install coverage docopt yarg requests
+
+            - name: Calculate coverage
+              run: coverage run --source=pipreqs -m unittest discover
+
+            - name: Create XML report
+              run: coverage xml
+
```

Comment:
Would you consider adding a report here? like it is done in [Code Coverage Report](https://github.com/marketplace/actions/code-coverage-report-action) using insightsengineering/coverage-action@v2. It would be nice to see the status during the PR progress.

Reply:
nice! I wanted something like that but at the time I wrote the task I hadnt found anything of this sort, will try to add some sort of report

Reply:
moved to issue #417